### PR TITLE
Updated renumerate function to remove hardcoded stage names

### LINKED ISSUE #58: Refactor renumerate_stages
https://github.com/gogoair/foremast/blob/master/src/foremast/pipeline/renumerate_stages.py

This can be refactored to be more agnostic to stage names. Right now it is kind of specific to the Gogo workflow but there should be a better way.

### PR DESCRIPTION
**Pull request by ndcampbell**
fixes #58 

I changed the logic to look for `"a"` or `"b"`  in the stage templates. Stages marked with `"a"` are mainline required pipeline stages. stages marked with `"b"` are secondary parallel stages. The order is already handled by https://github.com/gogoair/foremast/blob/master/src/foremast/templates/pipeline/pipeline_stages.json.j2 template.

This change would break existing templates, but makes the enumerate code a lot cleaner and supportable.

I tested this against our current pipeline and it works consistently.

### CODE REVIEW

**Comment by sijis** [comment]
I think it all makes sense. It would be more clear if `"a"` and `"b"` was something like `"required"` and `"optional"`. We will be sad 6 months from now and what what does "a" or "b" mean.

**Comment by ndcampbell** [comment]
I agree but am having a hard time with what key to use. `"optional"` is not the right word, the stages are still required, just no other stages depend on them. Maybe `"mainline"` and `"parallel"`?

**Comment by e4r7hbug** [comment]
Could go with Git naming conventions: master, merge, branch.

**Comment by ndcampbell** [comment]
Yeah, I kind of like `"master"`, `"branch"`, and `"merge"`. They are at least a good visual representation of the intended goal.

**Comment by sijis** [comment]
@ndcampbell @e4r7hbug I do like them with exception of `merge`. I don't think that's as clear as `master` and `branch`. I don't have an alternative naming, so go with that.

File: src/foremast/pipeline/renumerate_stages.py [hunk 1]
```diff
@@ -57,24 +34,18 @@ def renumerate_stages(pipeline):
...
-            stage['requisiteStageRefIds'] = [str(main_index)]
-            stage['refId'] = str(main_index * 102)
-        elif stage['type'] == 'bake' or idx == 0:
-            stage['requisiteStageRefIds'] = []
-            stage['refId'] = str(main_index)
-        else:
-            stage['requisiteStageRefIds'] = [str(main_index)]
+    main_index = 0
+    for stage in stages:
+        if stage['refId'].lower() == 'master':
```
Comment (e4r7hbug) [inline src/foremast/pipeline/renumerate_stages.py]: Save `stage['refId'].lower()` in a variable.

File: src/foremast/pipeline/renumerate_stages.py [hunk 2]
```diff
@@ -22,32 +22,9 @@
...
-    +---------+------------------------+------------+
-    | 4       | judgement              | 3          |
-    +---------+------------------------+------------+
-    | 5       | deploy stage           | 4          |
-    +---------+------------------------+------------+
-    | 6       | QE stage               | 5          |
-    +---------+------------------------+------------+
-    | 500     | git tagger stage       | 5          |
-    +---------+------------------------+------------+
+       stage order is defined in the templates.
```
Comment (e4r7hbug) [inline src/foremast/pipeline/renumerate_stages.py]: Docstring formatting.
```python
"""Renumber....

Stage order is defined in the templates. Use ``refId`` to determine dependency ordering.

    * ``a`` - Any...
    * ``b`` - Any...
"""
```

File: src/foremast/pipeline/renumerate_stages.py [hunk 3]
```diff
@@ -22,32 +22,9 @@
...
+       refId == 'a' - Any mainline required stage
```
Comment (e4r7hbug) [inline src/foremast/pipeline/renumerate_stages.py]: Update `a` and `b` to the new names we're trying out.

**Pull request merged**